### PR TITLE
Fully remove Navigation Editor screen and associated `edit-navigation` package

### LINKED ISSUE #43620: Remove the Navigation Editor Screen
We should remove the code that comprises the experimental Navigation Editor screen (i.e. `@wordpress/edit-navigation` package).

> @getdave, if the Navigation Screen has been removed, does that mean we could review and close pretty much all the open [related issues](https://github.com/WordPress/gutenberg/labels/%5BFeature%5D%20Navigation%20Screen)?

I think that's reasonable.

_Originally posted by @getdave in https://github.com/WordPress/gutenberg/issues/40878#issuecomment-1227694458_

### PR DESCRIPTION
**Pull request by getdave**
## What?

Removes the Navigation Editor screen from codebase and removes `@wordpress/edit-navigation` screen.

Closes https://github.com/WordPress/gutenberg/issues/43620

## Why?

For some time now this screen has been:
- removed from the Gutenberg UI
- deprecated 

As there are no plans to continue working on this concept it's time to remove it and the associated package.

Note that many learnings were taken from the work on this package which have since been applied to other projects surrounding Navigation. But...it's time to let it go 😢 

## How?

Removes all file references.

## Testing Instructions

- Check all editors work as expected
- Check for PHP errors
- Check codebase for references to the package or files.
- Check I removed the package correctly.

### CODE REVIEW

**Comment by noisysocks** [comment]
Makes sense to remove it. I imagine if the Navigation screen is picked back up that it'd be a completely different implementation that relies more heavily on the Navigation block and its block inspector. Plus can always look at git history to copy any code that we want to reuse.

Some of the test failures seem genuine. I think some E2E snapshots need to be updated.

Pinging @youknowriad in case he has any lessons to share about removing a package. (He removed `wp.nux` in https://github.com/WordPress/gutenberg/pull/46110.)

This package never made it into Core so there shouldn't be any backwards compatibility concerns about removing it.

If tests pass then this lgtm.

**Comment by getdave** [comment]
Yeh I'll sort those tests out soon. Thanks @noisysocks

**Comment by youknowriad** [comment]
Looks good to me, especially since this was never a core package 👍

**Comment by getdave** [comment]
Ok this is good to go 🤞 @scruffian can you ✅ based on the feedback from others above?

File: bin/plugin/commands/test/__snapshots__/changelog.js.snap [hunk 1]
```diff
@@ -43,6 +43,7 @@ exports[`getChangelog verify that the changelog is properly formatted 1`] = `
 - Fix block appender position in classic themes. ([33895](https://github.com/WordPress/gutenberg/pull/33895))
 - Fix misspelling of \\"queries\\" in filter documentation. ([33799](https://github.com/WordPress/gutenberg/pull/33799))
 - Fix positioning discrepancy with draggable chip. ([33893](https://github.com/WordPress/gutenberg/pull/33893))
+- Navigation Editor: Avoid React warning when creating a new menu. ([33843](https://github.com/WordPress/gutenberg/pull/33843))
```
Comment (draganescu) [inline bin/plugin/commands/test/__snapshots__/changelog.js.snap:46]: What is this line about in this PR?

**Review by scruffian — APPROVED** [review]
LGTM

−7,576!

**Review by draganescu — COMMENTED** [review]
I think we don't want to remove the entries in the changelogs since they serve as good history to refer to when bad ideas resurface 😁

**Comment by getdave** [comment]
# Dev note for WP 6.2

## Removal of the Navigation Editor screen and associated package

The Navigation Editor screen was an experimental feature of the Gutenberg Plugin that contributors worked on as a potential replacement for the classic `Menus` screen using a block-based paradigm. The code for this feature existed within a dedicated `@wordpress/edit-navigation` package.

As the Gutenberg Navigation project has evolved however, the requirement for a dedicated screen has become less pronounced, with features originally proposed for/included within the screen being incorporated into the Navigation block itself.

During the 6.1 cycle, due to lack of contributions, the experiment was informally deprecated in the Plugin by being removed from the `Experiments` screen in the Gutenberg Plugin settings page. 

With the arrival of list view editing for the Navigation block in WordPress 6.2, menus can now be edited away from the editor canvas, and therefore it is finally time to formally remove the experiment from the codebase.

The `@wordpress/edit-navigation` package has now been removed from the Gutenberg repository and then associated package unpublished from `npm`.

**Comment by getdave** [comment]
@talldan @draganescu Would you mind reviewing the above for accuracy/spelling/grammar .etc? Much appreciated.

**Comment by talldan** [comment]
It sounds good to me 👍

**Comment by bph** [comment]
Thanks for the Dev Note @getdave  - I added it to the [Miscellaneous Editor Changes in 6.2](https://make.wordpress.org/core/?p=102924&preview=1&_ppp=1f21a08839) post.  (Public post preview link)

**Pull request merged**